src/cli/commands/publish.js

```diff
@@ -127,6 +127,27 @@ export async function run(config: Config, reporter: Reporter, flags: Object, arg
     throw new MessageError(reporter.lang('unknownFolderOrTarball'));
   }
 
+  // save current version so we can restore it in case of an abort
+  const oldVersion = await getVersion(config);
+
+  // sorry if this try/catch breaks your stack trace :'(
```

Comment:
No need for this comment, rethrowing should keep the callstack more or less right